packages/davinci-client/src/lib/davinci.api.ts

```diff
@@ -216,7 +231,12 @@ export const davinciApi = createApi({
 
         const cacheEntry: DaVinciCacheEntry = api.getCacheEntry();
 
-        handleResponse(cacheEntry, api.dispatch, response?.status || 0);
+        handleResponse(
+          cacheEntry,
+          api.dispatch,
+          response?.status || 0,
+          (api.extra as Extras).logger,
```

Comment:
I see why we are doing this throughout the code, but something doesn't sit right with me making the logger an argument to all these functions.

it feels like the dependency leaks through all the code.

I don't have a great solution to propose for this unfortunately, maybe I have to sleep on it.

Reply:
This is to pass the same "instance" of the logger down into the important bits of the API. The only other solution is to have each low-level module initialize it's own logger, which I didn't really want to do, but we could do it.

It's worth noting that this `extra` property is only available to the `createApi` function of RTK, it's not accessible to the entire store and all the slices and reducers. So, it's very much intended for just the network layer, per se.